src/System.Private.CoreLib/shared/System/IO/Stream.cs

```diff
@@ -189,6 +189,73 @@ private int GetCopyBufferSize()
             return bufferSize;
         }
 
+        public virtual void CopyTo(ReadOnlySpanAction<byte, object> callback, object state, int bufferSize)
+        {
+            if (callback == null) throw new ArgumentNullException(nameof(callback));
+
+            CopyTo(new WriteCallbackStream(callback, state), bufferSize);
+        }
+
+        public virtual Task CopyToAsync(Func<ReadOnlyMemory<byte>, object, CancellationToken, Task> callback, object state, int bufferSize, CancellationToken cancellationToken)
+        {
+            if (callback == null) throw new ArgumentNullException(nameof(callback));
+
+            return CopyToAsync(new WriteCallbackStream(callback, state), bufferSize, cancellationToken);
+        }
+
+        private sealed class WriteCallbackStream : Stream
+        {
+            private readonly ReadOnlySpanAction<byte, object>? _action;
+            private readonly Func<ReadOnlyMemory<byte>, object, CancellationToken, Task>? _func;
+            private readonly object _state;
+
+            public WriteCallbackStream(ReadOnlySpanAction<byte, object> action, object state)
+            {
+                _action = action;
+                _state = state;
+            }
+
+            public WriteCallbackStream(Func<ReadOnlyMemory<byte>, object, CancellationToken, Task> func, object state)
+            {
+                _func = func;
+                _state = state;
+            }
+
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                if (_action == null) throw new NotSupportedException();
+                _action(new ReadOnlySpan<byte>(buffer, offset, count), _state);
+            }
+
+            public override void Write(ReadOnlySpan<byte> span)
+            {
+                if (_action == null) throw new NotSupportedException();
+                _action(span, _state);
+            }
+
+            public override Task WriteAsync(byte[] buffer, int offset, int length, CancellationToken cancellationToken)
+            {
+                if (_func == null) throw new NotSupportedException();
+                return _func(new ReadOnlyMemory<byte>(buffer, offset, length), _state, cancellationToken);
+            }
+
+            public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
+            {
+                if (_func == null) throw new NotSupportedException();
+                return new ValueTask(_func(buffer, _state, cancellationToken));
+            }
+
+            public override bool CanRead => false;
+            public override bool CanSeek => false;
+            public override bool CanWrite => true;
+            public override void Flush() { }
```

Comment:
This should override FlushAsync to nop as well. The base implementation will, too, just in a more expensive way.

Reply:
I agree. Fixed.